app/charts/chart-config-ui-options.ts

```diff
@@ -229,12 +230,14 @@ export const chartConfigOptionsUISpec: ChartSpecs = {
         field: "areaLayer",
         optional: true,
         componentTypes: ["GeoShapesDimension"],
+        allowOtherFieldComponents: true,
         filters: true,
       },
       {
         field: "symbolLayer",
         optional: true,
         componentTypes: ["GeoCoordinatesDimension", "GeoShapesDimension"],
+        allowOtherFieldComponents: true,
```

Comment:
I am not sure to understand clearly what this means, could you add a docstring on the field in the interface ?
What do you think of `exclusive: false` ?

```patch
- allowOtherFieldComponents?: boolean;
+ /** If false, using a dimension in this encoding will not prevent it to be used in an other encoding. Default: true */
+ exclusive?: boolean
```